### PR TITLE
File names in file header comments

### PR DESCRIPTION
**Pull request by StefRe**
Trying to familiarize with the source code I came across  some discrepancies between the file names in the file header comments and the actual file names. Maybe it's pedantic but I thought having the correct names might facilitate reading the code especially when using printouts.

Some files don't appear to have file header comments and boilerplate license statements as recommended by the [Style guide for Google-originated open-source projects](https://github.com/google/styleguide/blob/gh-pages/cppguide.html#L4537), but I don't feel that I were in a position to supply this information. This pertains to the following files:

```
ccutil\ccutil.cpp
neural_networks\runtime\*.cpp
opencl\openclwrapper.cpp
textord\gap_map.cpp
textord\tospace.cpp
training\commandlineflags.cpp
training\set_unicharset_properties.cpp
viewer\svpaint.cpp
```

### CODE REVIEW

File: ccstruct/coutln.h [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**********************************************************************
- * File:						coutln.c      (Formerly:  coutline.c)
+ * File:	coutln.h      (Formerly:  coutline.c)
```
Comment (amitdo) [inline ccstruct/coutln.h]: `Formerly coutline.c` should be removed.

**Comment by jbreiden** [comment]
Tesseract predates Google.

File: ccutil/serialis.cpp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /**********************************************************************
- * File:        serialis.h  (Formerly serialmac.h)
+ * File:        serialis.cpp  (Formerly serialmac.h)
```
Comment (amitdo) [inline ccutil/serialis.cpp]: `Formerly serialmac.h` should be replaced with `Formerly serialmac.c`

File: ccstruct/coutln.h (hunk 1 above)
Comment (amitdo) [inline ccstruct/coutln.h]: `Formerly coutline.c` should be removed.

File: ccutil/bits16.cpp [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /**********************************************************************
- * File:        bits16.h  (Formerly bits8.h)
+ * File:        bits16.cpp  (Formerly bits8.h)
```
Comment (amitdo) [inline ccutil/bits16.cpp]: `Formerly bits8.h` should be replaced with `Formerly bits8.c`

**Comment by amitdo** [comment]
>Trying to familiarize with the source code I came across some discrepancies between the file names in the file header comments and the actual file names. Maybe it's pedantic but I thought having the correct names might facilitate reading the code especially when using printouts.

+1 for the suggestion to fix it!

but because your patches touch too many files and for other reasons I prefer that a core developer from Google to take it.

File: cube/char_set.cpp [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**********************************************************************
- * File:        char_samp_enum.cpp
```
Comment (stweil) [inline cube/char_set.cpp]: I'd omit the comment in parentheses (who cares for the former name?).

File: cube/char_set.h [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /**********************************************************************
- * File:        char_samp_enum.h
```
Comment (stweil) [inline cube/char_set.h]: Same here.

**Comment by stweil** [comment]
Fixing the wrong or out-dated names is good. I suggest to remove all modifications which are not related to such names and then apply this PR. Even if the PR were applied as it is, it would improve the current situation.

File: cube/feature_hybrid.h [hunk 6]
```diff
@@ -19,7 +19,7 @@
 
 // The FeatureHybrid class implements a Bitmap feature extractor class. It
 // inherits from the FeatureBase class
-// This class describes the a hybrid feature vector composed by combining
```
Comment (stweil) [inline cube/feature_hybrid.h]: This modification is unrelated to the commit comments.

File: cube/char_set.cpp (hunk 4 above)
Comment (StefRe) [inline cube/char_set.cpp]: I added these old file names just for uniformity's sake to be consistent with the other files.

File: cube/feature_hybrid.h (hunk 6 above)
Comment (StefRe) [inline cube/feature_hybrid.h]: OK, you're right. Just subconsciously corrected this obvious typo when reading the comment.

File: cube/char_set.cpp (hunk 4 above)
Comment (stweil) [inline cube/char_set.cpp]: Such comments might have been useful 20 years ago, but nowadays we have configuration management and can track all history in Git, so I think there is no need for such comments in the code any more. I'd prefer removing the existing comments (in a later PR of course) instead of adding more of them to reach uniformity.

**Comment by zdenop** [comment]
this PR can not be merged to current code. Can you please update it?

**Comment by amitdo** [comment]
I think @theraysmith should say if he is OK with these changes.
Ray, also see @stweil https://github.com/tesseract-ocr/tesseract/pull/344#issuecomment-239267937

 @jbreiden said something, but for some reason deleted his comment.

**Comment by theraysmith** [comment]
The request is reasonable, but I would rather spend my limited time fixing actual bugs than up-integrating/merging such large changes (and asking colleagues to review them) that perform little real useful purpose, especially when there is a chance that a large chunk of the code could go away in the near future.
